Add email to user profile view

Senorsen opened the pull request:
Current version of GitLab doesn't display user email on user profile page, which is very inconvenient.

This commit added email field to user profile view. 

Before
![screenshot from 2015-04-11 20 35 31](https://cloud.githubusercontent.com/assets/4176091/7101394/99bd4346-e08a-11e4-82f9-c0ecc7d5203e.png)

After
![screenshot from 2015-04-11 20 35 21](https://cloud.githubusercontent.com/assets/4176091/7101396/a2e53bc2-e08a-11e4-8843-eb21a9976adc.png)

Razer6 commented:
:-1: For privacy. I want to control if I I want to show my email in public.

rumpelsepp commented:
>  For privacy.

Remember: If you have public projects your email is already there in the commit meta data.

Senorsen commented:
Let me try...

Senorsen commented:
@Razer6 
Hi, I added a checkbox in the profile settings page, users controller will do some check, determine whether to display this user's email.

![screenshot from 2015-04-11 23 17 36](https://cloud.githubusercontent.com/assets/4176091/7101913/f7940520-e0a0-11e4-8173-03b91f692c89.png)

Razer6 commented on db/migrate/20150411214929_add_email_display_in_profile.rb:
```diff
@@ -0,0 +1,5 @@
+class AddEmailDisplayInProfile < ActiveRecord::Migration
+  def change
+    add_column :users, :email_display_in_profile, :boolean, default: true, null: false
```
I won't do a default `true` here

Razer6 commented:
@rumpelsepp Yes and no. One could have a different mailing address for the commits... @Senorsen Can you rebase? Seems some commits not from you are popping up.

Senorsen committed:
Add email to user profile view, and fix the failed test

Add email to user profile, and add an option, which will
allow user to select whether they wants to display main email
address in their profile page.

The user himself / herself or administrators can always see
this email address.

Senorsen commented:
@Razer6 Hi, I have rebased and fixed some failed tests, it seems everything is ok now.

Senorsen commented:
@Razer6 @randx Hi, I'd like to know whether this merge request can be merged. I'm looking forward to your reply!

randx commented:
I would like to see separate field for public email

rumpelsepp commented:
> I would like to see separate field for public email
:+1:

Senorsen commented:
@randx Just like the way GitHub currently uses? Seems good...

randx commented:
@Senorsen yes why not

Senorsen committed:
Allow user to choose which email to be pulic

This commit allows user to show one of their emails in profile page,
or don't show email in this page.
